controllers/iamauthpolicy_controller.go

```diff
@@ -249,13 +275,50 @@ func (c IAMAuthPolicyController) updatePolicyCondition(ctx context.Context, k8sP
 		status = metav1.ConditionFalse
 	}
 	cnd := metav1.Condition{
-		Type:               string(gwv1alpha2.PolicyConditionAccepted),
-		Status:             status,
-		ObservedGeneration: k8sPolicy.Generation,
-		LastTransitionTime: metav1.Now(),
-		Reason:             string(reason),
+		Type:   string(gwv1alpha2.PolicyConditionAccepted),
+		Status: status,
+		Reason: string(reason),
 	}
 	meta.SetStatusCondition(&k8sPolicy.Status.Conditions, cnd)
```

Comment:
Off topic: How you figured out to use `meta.SetStatusCondition()`? Seems elsewhere in our repo should also it to update status...

https://github.com/aws/aws-application-networking-k8s/blob/a2d5b973b3516f11784d1a764830e8de2eadf4a7/controllers/accesslogpolicy_controller.go#L302

https://github.com/aws/aws-application-networking-k8s/blob/cd77f5d506cc9ecc354dc46460f0364188113129/pkg/utils/condition.go#L7

Reply:
I searched for the utility methods and found it. I dont think we need our own utility functions.

Reply:
Great, I created a issue for it https://github.com/aws/aws-application-networking-k8s/issues/459